docs/DesignDocs/CustomAttestation_V3.md

```diff
@@ -0,0 +1,433 @@
+Custom Attestation Data Formats for Open Enclave V3 Updates
+================
+
+As custom attestation plugins are developed and features are requested, new APIs were added to:
+
+1. Query built-in SGX attesters and verifiers to be registered.
+2. Select a format of the evidence that the registered attesters can generate.
+3. Query evidence parameters to be used as optional inputs when generating evidences.
+
+The following API was affected:
+
+1. Built-in SGX attesters and verifiers
+
+This document describes the added and affected APIs as well as keeping the existing support for version 2 of the API. This document assumes the user is familiar with the SGX attestation and [custom attestation](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/CustomAttestation.md) with plugins.
```

Comment:
I would recommend marking the new APIs as @experimental

Reply:
Good point.

---
In reply to: [394649764](https://github.com/openenclave/openenclave/pull/2621#discussion_r394649764) [](ancestors = 394649764)